src/sb_pointcloud_processing/src/pcl_transform.cpp

```diff
@@ -14,38 +14,56 @@
 #include <sb_utils.h>
 
 ros::Publisher pub;
-std::string output_frame;
 tf2_ros::Buffer tf_buffer;
 
+std::string output_frame;
+
+// The amount of time (s) for a new pointcloud transform to be published.
+double transform_period;
+
 void pointCloudCallback(const sensor_msgs::PointCloud2::ConstPtr& msg) {
     try {
         // Create an empty pointcloud
         sensor_msgs::PointCloud2 output;
         // Transform the pointcloud to the requested frame
         geometry_msgs::TransformStamped tf_stamped = tf_buffer.lookupTransform(
-        msg->header.frame_id, output_frame, ros::Time(0), ros::Duration(1.0));
+        msg->header.frame_id, output_frame, ros::Time(0), ros::Duration(0.1));
+
         tf2::doTransform(*msg, output, tf_stamped);
 
+        output.header.frame_id = output_frame;
+
         // Publish the transformed pointcloud
         pub.publish(output);
-    } catch (tf2::TransformException ex){
+
+        ros::Duration(transform_period).sleep();
+    } catch (tf2::TransformException ex) {
         ROS_WARN("%s", ex.what());
         ros::Duration(1.0).sleep();
```

Comment:
Hmm, does it make sense to sleep here? That means if there is a TF issue on one frame, we won't transform anything for 1 second? I think this is here to make sure we don't overwhelm the log with messages, so perhaps a timer-based system so that we only _log_ every 500ms or so might be a better option?

Reply:
What do you mean by this? Do you mean to change the sleep duration to 0.5s? Or do you mean create a timer callback that logs the status of the system every 0.5s?

Reply:
I mean remove the `sleep` here so that we're still trying to transform things as fast as possible, but add a check to see if >1s has passed since the last time we logged so that we don't overwhelm the log by logging 30 times a second (or whatever the framerate is)